worker/lease/manager_cross_test.go

```diff
@@ -0,0 +1,313 @@
+		c.Assert(err, gc.Equals, corelease.ErrClaimDenied)
+	})
+}
+
+func (s *CrossSuite) TestClaimAcrossModels(c *gc.C) {
+	lease1 := key("ns", "m1", "summerisle")
+	lease2 := key("ns", "m2", "summerisle")
+	fix := Fixture{
+		expectCalls: []call{{
+			method: "ClaimLease",
+			args: []interface{}{
+				lease1,
+				corelease.Request{"sgt-howie", time.Minute},
+			},
+			callback: func(leases map[corelease.Key]corelease.Info) {
+				leases[lease1] = corelease.Info{
+					Holder: "sgt-howie",
+					Expiry: offset(time.Second),
+				}
+			},
+		}, {
+			method: "ClaimLease",
+			args: []interface{}{
+				lease2,
+				corelease.Request{"rowan", time.Minute},
+			},
+			callback: func(leases map[corelease.Key]corelease.Info) {
+				leases[lease2] = corelease.Info{
+					Holder: "rowan",
+					Expiry: offset(time.Second),
+				}
+			},
+		}},
+	}
+	fix.RunTest(c, func(manager *lease.Manager, _ *testing.Clock) {
+		claimer1, err := manager.Claimer("ns", "m1")
+		c.Assert(err, jc.ErrorIsNil)
+		claimer2, err := manager.Claimer("ns", "m2")
+		c.Assert(err, jc.ErrorIsNil)
+
+		err = claimer1.Claim("summerisle", "sgt-howie", time.Minute)
+		c.Assert(err, jc.ErrorIsNil)
+		err = claimer2.Claim("summerisle", "rowan", time.Minute)
+		c.Assert(err, jc.ErrorIsNil)
+
+		err = claimer1.Claim("summerisle", "lord-summerisle", time.Minute)
+		c.Assert(err, gc.Equals, corelease.ErrClaimDenied)
+	})
+}
+
+func (s *CrossSuite) TestWaitAcrossNamespaces(c *gc.C) {
+	lease1 := key("ns1", "model", "summerisle")
+	lease2 := key("ns2", "model", "summerisle")
+	fix := Fixture{
+		leases: map[corelease.Key]corelease.Info{
+			lease1: {
+				Holder: "sgt-howie",
+				Expiry: offset(time.Second),
+			},
+			lease2: {
+				Holder: "willow",
+				Expiry: offset(time.Minute),
+			},
+		},
+		expectCalls: []call{{
+			method: "Refresh",
+		}, {
+			method: "ExpireLease",
+			args:   []interface{}{lease1},
+			callback: func(leases map[corelease.Key]corelease.Info) {
+				delete(leases, lease1)
+			},
+		}},
+	}
+	fix.RunTest(c, func(manager *lease.Manager, clock *testing.Clock) {
+		b1 := newBlockTest(manager, lease1)
+		b2 := newBlockTest(manager, lease2)
+
+		b1.assertBlocked(c)
+		b2.assertBlocked(c)
+
+		clock.Advance(time.Second)
+
+		err := b1.assertUnblocked(c)
+		c.Assert(err, jc.ErrorIsNil)
+		b2.assertBlocked(c)
+	})
+}
+
+func (s *CrossSuite) TestWaitAcrossModels(c *gc.C) {
+	lease1 := key("ns", "m1", "summerisle")
+	lease2 := key("ns", "m2", "summerisle")
+	fix := Fixture{
+		leases: map[corelease.Key]corelease.Info{
+			lease1: {
+				Holder: "sgt-howie",
+				Expiry: offset(time.Second),
+			},
+			lease2: {
+				Holder: "willow",
+				Expiry: offset(time.Minute),
+			},
+		},
+		expectCalls: []call{{
+			method: "Refresh",
+		}, {
+			method: "ExpireLease",
+			args:   []interface{}{lease1},
+			callback: func(leases map[corelease.Key]corelease.Info) {
+				delete(leases, lease1)
+			},
+		}},
+	}
+	fix.RunTest(c, func(manager *lease.Manager, clock *testing.Clock) {
```

Comment:
this could be a common helper for wait across namespaces and wait across models

Reply:
Yeah, good call.